zerver/lib/upload.py

```diff
@@ -452,10 +456,15 @@ def get_realm_icon_url(self, realm_id: int, version: int) -> str:
         # ?x=x allows templates to append additional parameters with &s
         return "https://%s.s3.amazonaws.com/%s/realm/icon.png?version=%s" % (bucket, realm_id, version)
 
-    def upload_realm_logo_image(self, logo_file: File, user_profile: UserProfile) -> None:
+    def upload_realm_logo_image(self, logo_file: File, user_profile: UserProfile,
+                                night_mode_logo: bool) -> None:
         content_type = guess_type(logo_file.name)[0]
         bucket_name = settings.S3_AVATAR_BUCKET
-        s3_file_name = os.path.join(str(user_profile.realm.id), 'realm', 'logo')
+        if night_mode_logo is True:
```

Comment:
@timabbott  I have made changes in which i have passed a data value `night: true` but it is not functioning properly as changes are also required in this file but I am not able to understand how to do it. Can you please help ?